ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/transform/PubsubMessageToObjectNode.java

```diff
@@ -361,22 +362,30 @@ private void processField(String jsonFieldName, Field field, JsonNode value, Obj
             if (!Json.isNullOrEmpty(props)) {
               repeatedAdditionalProperties.add(props.get(jsonFieldName));
             } else {
-              repeatedAdditionalProperties.addNull();
+              repeatedAdditionalProperties.addObject();
             }
           }
         } else {
+          ArrayNode filteredValue = Json.createArrayNode();
           for (JsonNode record : value) {
             final ObjectNode props = additionalProperties == null ? null : Json.createObjectNode();
-            transformForBqSchema((ObjectNode) record, field.getSubFields(), props);
```

Comment:
Based on my understanding of the existing code, nulls within object arrays should hit this line and throw an exception since `record` will be `NullNode` rather than `ObjectNode`. Is it possible that these are currently going to error output?

Reply:
there is no error output for the new sink. it will instead nack the message in pubsub, causing the sink to retry indefinitely, unless the error is transient or gets fixed in an ingestion-sink update.